plugin/gcp/instance.py

```diff
@@ -0,0 +1,239 @@
+# #######
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+from plugin.gcp import utils
+from plugin.gcp.service import GoogleCloudPlatform
+from plugin.gcp.service import GCPError
+from plugin.gcp.service import blocking
+
+
+class Instance(GoogleCloudPlatform):
+    ACCESS_CONFIG = 'External NAT'
+    ACCESS_CONFIG_TYPE = 'ONE_TO_ONE_NAT'
+    NETWORK_INTERFACE = 'nic0'
+    STANDARD_MACHINE_TYPE = 'n1-standard-1'
+
+    def __init__(self,
+                 config,
+                 logger,
+                 instance_name,
+                 image=None,
```

Comment:
ok, I want to revise me comment :smile: 
I prefer only incuding the config and logger - which are required to initialize the super class - in the init method. Everything else - instance name, image, machine_type etc should be passed directly to the methods ("create", "list") when required. In each method declaration, mandatory args (such as "image" passed to "create"), should not have the default None.